src/Pretzel/Program.cs

```diff
@@ -1,89 +1,160 @@
+            },
+            new Option(new[] { "--source", "-source", "-s", "--s" }, "The path to the source site (default current directory)")
+            {
+                Argument = new Argument<string>(() => fileSystem.Directory.GetCurrentDirectory())
+            }
+        };
+#pragma warning restore S2223 // Non-constant static fields should not be visible
 
-        [Export("SourcePath")]
-        public string SourcePath { get; }
+        [Import]
+        public CommandCollection CommandCollection { get; set; }
 
-        private BaseParameters parameters { get; }
+        [Export]
+        public IFileSystem FileSystem { get; set; } = fileSystem;
 
-        public Program()
+        private static async Task<int> Main(string[] args)
         {
-            parameters = BaseParameters.Parse(Args, FileSystem);
-            SourcePath = parameters.Path;
-        }
+            var rootCommand = new RootCommand
+            {
+                TreatUnmatchedTokensAsErrors = false
+            };
 
-        private static string[] Args;
-        private static void Main(string[] args)
-        {
-            Args = args;
-            var program = new Program();
+            args = PatchSourcePath(args);
 
-            program.InitializeTrace();
-            Tracing.Info("starting pretzel...");
-            Tracing.Debug("V{0}", Assembly.GetExecutingAssembly().GetName().Version);
+            foreach (var option in GlobalOptions)
+            {
+                rootCommand.AddOption(option);
+            }
 
-            using (program.Compose())
+            rootCommand.Handler = CommandHandler.Create(async (bool debug, bool safe, string source) =>
             {
+                try
+                {
+                    InitializeTrace(debug);
+                    Tracing.Info("starting pretzel...");
+                    Tracing.Debug("V{0}", Assembly.GetExecutingAssembly().GetName().Version);
 
-                if (program.parameters.Help || !args.Any())
+                    using (var host = Compose(debug, safe, source))
+                    {
+                        var program = host.GetExport<Program>();
+                        var result = await program.Run(GlobalOptions, args);
+                        WaitForClose();
+                        return result;
+                    }
+                }
+                catch (Exception ex)
                 {
-                    program.ShowHelp();
-                    return;
+                    Tracing.Error(ex.Message);
+                    WaitForClose();
+                    return -1;
                 }
+            });
+            try
+            {
+                return await rootCommand.InvokeAsync(args);
+            }
+            catch (Exception ex)
+            {
+                Tracing.Error(ex.Message);
+                WaitForClose();
+                return -1;
+            }
+        }
+
+        internal static string[] PatchSourcePath(string[] args)
```

Comment:
I'm not sure if `Arguments` can be used in this case. The reason why this exist is: in previous versions it was possible to write: `pretzel bake c:\path\to\blog` as well as `pretzel bake -s c:\path\to\blog`. If they can be optional it may be possible, but I have no idea how.

Reply:
I haven't succeeded on using an Argument on a RootCommand so this will do.